src/winter_onboarding_2021/fleet_management_service/db/cab.clj

```diff
@@ -1,44 +1,64 @@
 (ns winter-onboarding-2021.fleet-management-service.db.cab
   (:require [winter-onboarding-2021.fleet-management-service.db.core :as db]
             [honey.sql :as sql]
+            [winter-onboarding-2021.fleet-management-service.specs :as specs]
+            [clojure.spec.alpha :as s]
+            [winter-onboarding-2021.fleet-management-service.error :as errors]
+            [winter-onboarding-2021.fleet-management-service.utils :as utils]
             [honey.sql.helpers :as h :refer [select from limit offset order-by update set where]]))
 
-(defn create [cabs]
-  (db/insert! :cabs cabs))
+(defn create [cab]
+  (if (s/valid? ::specs/cabs cab)
+    (db/insert! :cabs cab)
+    errors/validation-failed))
 
 (defn select! [off lim]
-  (db/query! (sql/format (-> (select :id
-                                     :name
-                                     :licence-plate
-                                     :distance-travelled
-                                     :created-at
-                                     :updated-at)
-                             (from :cabs)
-                             (limit lim)
-                             (offset off)
-                             (order-by :created-at)))))
+  (if (s/valid? ::specs/pagination-params {:offset off :limit lim})
+    (db/query! (sql/format (-> (select :id
+                                       :name
+                                       :licence-plate
+                                       :distance-travelled
+                                       :created-at
+                                       :updated-at)
+                               (from :cabs)
+                               (limit lim)
+                               (offset off)
+                               (order-by :created-at))))
+    errors/validation-failed))
 
 (defn get-count []
   (db/query! (sql/format (-> (select [:%count.* :count])
                              (from :cabs)))))
 
 (defn get-by-id [id]
-  {:pre [(uuid? id)]}
-  (db/get-by-id! :cabs id))
+  (if (uuid? id)
+    (db/get-by-id! :cabs id)
+    errors/id-not-uuid))
 
 (defn find-by-keys [key-map]
-  (db/find-by-keys! :cabs key-map))
+  (let [valid-key-map (utils/select-keys-from-spec key-map 
+                                                   ::specs/cabs-all-attr)]
+    (if (empty? valid-key-map)
+      errors/no-valid-keys
+      (db/find-by-keys! :cabs valid-key-map))))
 
 (defn update! [id cab]
-  {:pre [(uuid? id)]}
-  (db/query! (sql/format (-> (update :cabs)
-                             (set cab)
-                             (where [:= :id id])))))
+  (if (uuid? id)
```

Comment:
You can use a spec here as well.
[Opinion] This would be more readable using `cond`